go/logic/applier.go

```diff
@@ -211,86 +233,140 @@ func (this *Applier) CreateChangelogTable() error {
 	if err := this.DropChangelogTable(); err != nil {
 		return err
 	}
-	query := fmt.Sprintf(`create /* gh-ost */ table %s.%s (
-			id bigint auto_increment,
-			last_update timestamp not null DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
-			hint varchar(64) charset ascii not null,
-			value varchar(4096) charset ascii not null,
-			primary key(id),
-			unique key hint_uidx(hint)
-		) auto_increment=256
-		`,
-		sql.EscapeName(this.migrationContext.DatabaseName),
+	query := `create /* gh-ost */ table %s.%s (
+		id bigint auto_increment,
+		last_update timestamp not null DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+		hint varchar(64) charset ascii not null,
+		value varchar(4096) charset ascii not null,
+		primary key(id),
+		unique key hint_uidx(hint)
+	) auto_increment=256
+	`
+	dstQuery := fmt.Sprintf(query,
+		sql.EscapeName(this.migrationContext.DstDatabaseName),
+		sql.EscapeName(this.migrationContext.GetChangelogTableName()),
+	)
+	log.Infof("Creating changelog table %s.%s on dst",
+		sql.EscapeName(this.migrationContext.DstDatabaseName),
+		sql.EscapeName(this.migrationContext.GetChangelogTableName()),
+	)
+	if _, err := sqlutils.ExecNoPrepare(this.dstDB, dstQuery); err != nil {
+		return err
+	}
+	log.Infof("Changelog table created on dst")
+
+	if this.migrationContext.SrcIsAlsoDst() {
+		return nil
+	}
+
+	log.Infof("Creating changelog table %s.%s on src",
+		sql.EscapeName(this.migrationContext.DstDatabaseName),
```

Comment:
this.migrationContext.DstDatabaseName seems should be this.migrationContext.SrcDatabaseName